nash-native-client/src/ws_client/client.rs

```diff
@@ -578,6 +575,124 @@ impl Client {
     }
 }
 
+
+// Adding another layer of abstraction on top of client make it easier/safer to manage concurrent usage of the client
+// In particular this is necessary to spawn a process in the background that signs state
+pub struct Client {
+    inner: Arc<Mutex<InnerClient>>,
+    sign_loop_status: Arc<Mutex<bool>>,
```

Comment:
Where are we reading this value?
Can it be made Arc<AtomicBool>?

Reply:
yep i think so, good suggestion

Reply:
i actually don't think the value can be mutated easily across threads, so leaving it be for now. can do a small PR later if I'm wrong about that (very possible)